src/core/sync/mutex.d

```diff
@@ -170,8 +171,9 @@ class Mutex :
         else version( Posix )
         {
             int rc = pthread_mutex_unlock( &m_hndl );
+            static immutable syncEx = new SyncError("Unable to unlock mutex");
             if( rc )
-                throw new SyncError( "Unable to unlock mutex" );
+                throw syncEx;
```

Comment:
Yeah, that was the other "solution" to this that I found - but both seem to be giant hacks so I thought I'd just submit a PR and see what people better than I am in D think should be done here.

Thanks for the link, I'll look it over.